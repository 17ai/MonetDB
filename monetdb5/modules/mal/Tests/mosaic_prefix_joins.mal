b:= bat.new(:oid,:int);
bat.append(b,50);
bat.append(b,51);
bat.append(b,51);
bat.append(b,51);
bat.append(b,51);
bat.append(b,51);
bat.append(b,51);
bat.append(b,60);
bat.append(b,67);
bat.append(b,67);
bat.append(b,67);
bat.append(b,67);
bat.append(b,67);
bat.append(b,67);
bat.append(b,67);
bat.append(b,255);
bat.append(b,256);
bat.append(b,256);
bat.append(b,256);
bat.append(b,256);
bat.append(b,256);
bat.append(b,255);
bat.append(b,255);
bat.append(b,255);
bat.append(b,255);
bat.append(b,255);
base:=algebra.copy(b);

io.print(b);
x:= mosaic.compress(b,"prefix");
#mosaic.dump(x);
z:= mosaic.decompress(x);
io.print(z);
# Linear range compressions

c:= bat.new(:oid,:oid);
bat.append(c,0@0);
bat.append(c,2@0);
bat.append(c,4@0);
io.print(c);

#mosaic.dump(x);
s:= algebra.leftfetchjoin(c,base);
io.print(s);
xs:= mosaic.leftfetchjoin(c,x);
io.print(xs);

d:= bat.new(:oid,:int);
bat.append(d,50);
bat.append(d,67);
bat.append(d,nil:int);

(jl,jr):= algebra.join(base,d);
io.print(jl,jr);
(ml,mr):= mosaic.join(x,d);
io.print(ml,mr);

(jl,jr):= algebra.join(d,base);
io.print(jl,jr);
(ml,mr):= mosaic.join(d,x);
io.print(ml,mr);
