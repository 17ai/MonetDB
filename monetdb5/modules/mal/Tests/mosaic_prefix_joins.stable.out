stdout of test 'mosaic_prefix_joins` in directory 'monetdb5/modules/mal` itself:


# 12:33:36 >  
# 12:33:36 >  "mserver5" "--debug=10" "--set" "gdk_nr_threads=0" "--set" "mapi_open=true" "--set" "mapi_port=33381" "--set" "mapi_usock=/var/tmp/mtest-25823/.s.monetdb.33381" "--set" "monet_prompt=" "--forcemito" "--set" "mal_listing=2" "--dbpath=/export/scratch1/mk/mosaic//Linux/var/MonetDB/mTests_monetdb5_modules_mal" "mosaic_prefix_joins.mal"
# 12:33:36 >  

# MonetDB 5 server v11.20.0
# This is an unreleased version
# Serving database 'mTests_monetdb5_modules_mal', using 8 threads
# Compiled for x86_64-unknown-linux-gnu/64bit with 64bit OIDs and 128bit integers dynamically linked
# Found 15.590 GiB available main-memory.
# Copyright (c) 1993-July 2008 CWI.
# Copyright (c) August 2008-2014 MonetDB B.V., all rights reserved
# Visit http://www.monetdb.org/ for further information
# Listening for connection requests on mapi:monetdb://vienna.ins.cwi.nl:33381/
# Listening for UNIX domain connection requests on mapi:monetdb:///var/tmp/mtest-25823/.s.monetdb.33381
# MonetDB/GIS module loaded
# MonetDB/SQL module loaded
function user.main():void;
    b := bat.new(:oid,:int);
    bat.append(b,50);
    bat.append(b,51);
    bat.append(b,51);
    bat.append(b,51);
    bat.append(b,51);
    bat.append(b,51);
    bat.append(b,51);
    bat.append(b,60);
    bat.append(b,67);
    bat.append(b,67);
    bat.append(b,67);
    bat.append(b,67);
    bat.append(b,67);
    bat.append(b,67);
    bat.append(b,67);
    bat.append(b,255);
    bat.append(b,256);
    bat.append(b,256);
    bat.append(b,256);
    bat.append(b,256);
    bat.append(b,256);
    bat.append(b,255);
    bat.append(b,255);
    bat.append(b,255);
    bat.append(b,255);
    bat.append(b,255);
    base := algebra.copy(b);
    io.print(b);
    x := mosaic.compress(b,"prefix");
#mosaic.dump(x); 
    z := mosaic.decompress(x);
    io.print(z);
# Linear range compressions 
    c := bat.new(:oid,:oid);
    bat.append(c,0@0);
    bat.append(c,2@0);
    bat.append(c,4@0);
    io.print(c);
#mosaic.dump(x); 
    s := algebra.leftfetchjoin(c,base);
    io.print(s);
    xs := mosaic.leftfetchjoin(c,x);
    io.print(xs);
    d := bat.new(:oid,:int);
    bat.append(d,50);
    bat.append(d,67);
    bat.append(d,nil:int);
    (jl,jr) := algebra.join(base,d);
    io.print(jl,jr);
    (ml,mr) := mosaic.join(x,d);
    io.print(ml,mr);
    (jl,jr) := algebra.join(d,base);
    io.print(jl,jr);
    (ml,mr) := mosaic.join(d,x);
    io.print(ml,mr);
end user.main;
#--------------------------#
# h	t  # name
# void	int  # type
#--------------------------#
[ 0@0,	50  ]
[ 1@0,	51  ]
[ 2@0,	51  ]
[ 3@0,	51  ]
[ 4@0,	51  ]
[ 5@0,	51  ]
[ 6@0,	51  ]
[ 7@0,	60  ]
[ 8@0,	67  ]
[ 9@0,	67  ]
[ 10@0,	67  ]
[ 11@0,	67  ]
[ 12@0,	67  ]
[ 13@0,	67  ]
[ 14@0,	67  ]
[ 15@0,	255  ]
[ 16@0,	256  ]
[ 17@0,	256  ]
[ 18@0,	256  ]
[ 19@0,	256  ]
[ 20@0,	256  ]
[ 21@0,	255  ]
[ 22@0,	255  ]
[ 23@0,	255  ]
[ 24@0,	255  ]
[ 25@0,	255  ]
#--------------------------#
# h	t  # name
# void	int  # type
#--------------------------#
[ 0@0,	50  ]
[ 1@0,	51  ]
[ 2@0,	51  ]
[ 3@0,	51  ]
[ 4@0,	51  ]
[ 5@0,	51  ]
[ 6@0,	51  ]
[ 7@0,	60  ]
[ 8@0,	67  ]
[ 9@0,	67  ]
[ 10@0,	67  ]
[ 11@0,	67  ]
[ 12@0,	67  ]
[ 13@0,	67  ]
[ 14@0,	67  ]
[ 15@0,	255  ]
[ 16@0,	256  ]
[ 17@0,	256  ]
[ 18@0,	256  ]
[ 19@0,	256  ]
[ 20@0,	256  ]
[ 21@0,	255  ]
[ 22@0,	255  ]
[ 23@0,	255  ]
[ 24@0,	255  ]
[ 25@0,	255  ]
#--------------------------#
# h	t  # name
# void	oid  # type
#--------------------------#
[ 0@0,	0@0  ]
[ 1@0,	2@0  ]
[ 2@0,	4@0  ]
#--------------------------#
# h	t  # name
# void	int  # type
#--------------------------#
[ 0@0,	50  ]
[ 1@0,	51  ]
[ 2@0,	51  ]
#--------------------------#
# h	t  # name
# void	int  # type
#--------------------------#
[ 0@0,	50  ]
[ 1@0,	51  ]
[ 2@0,	51  ]
#--------------------------#
# h	t	t  # name
# void	oid	oid  # type
#--------------------------#
[ 0@0,	0@0,	0@0  ]
[ 1@0,	8@0,	1@0  ]
[ 2@0,	9@0,	1@0  ]
[ 3@0,	10@0,	1@0  ]
[ 4@0,	11@0,	1@0  ]
[ 5@0,	12@0,	1@0  ]
[ 6@0,	13@0,	1@0  ]
[ 7@0,	14@0,	1@0  ]
#--------------------------#
# h	t	t  # name
# void	oid	oid  # type
#--------------------------#
[ 0@0,	0@0,	0@0  ]
[ 1@0,	8@0,	1@0  ]
[ 2@0,	9@0,	1@0  ]
[ 3@0,	10@0,	1@0  ]
[ 4@0,	11@0,	1@0  ]
[ 5@0,	12@0,	1@0  ]
[ 6@0,	13@0,	1@0  ]
[ 7@0,	14@0,	1@0  ]
#--------------------------#
# h	t	t  # name
# void	oid	oid  # type
#--------------------------#
[ 0@0,	0@0,	0@0  ]
[ 1@0,	1@0,	8@0  ]
[ 2@0,	1@0,	9@0  ]
[ 3@0,	1@0,	10@0  ]
[ 4@0,	1@0,	11@0  ]
[ 5@0,	1@0,	12@0  ]
[ 6@0,	1@0,	13@0  ]
[ 7@0,	1@0,	14@0  ]
#--------------------------#
# h	t	t  # name
# void	oid	oid  # type
#--------------------------#
[ 0@0,	0@0,	0@0  ]
[ 1@0,	1@0,	8@0  ]
[ 2@0,	1@0,	9@0  ]
[ 3@0,	1@0,	10@0  ]
[ 4@0,	1@0,	11@0  ]
[ 5@0,	1@0,	12@0  ]
[ 6@0,	1@0,	13@0  ]
[ 7@0,	1@0,	14@0  ]

# 12:33:36 >  
# 12:33:36 >  "Done."
# 12:33:36 >  

